ASProgressPopUpView/ASProgressPopUpView.m

```diff
@@ -99,12 +100,24 @@ - (void)setPopUpViewColor:(UIColor *)color
     _popUpViewColor = color;
     _popUpViewAnimatedColors = nil; // animated colors should be discarded
     [self.popUpView setColor:color];
-    [self setGradientColors:@[color, color] withPositions:nil];
+    [self setGradientColors:@[color, color] withPositions:@[@(0.0), @(1.0)]];
 }
 
 - (void)setPopUpViewAnimatedColors:(NSArray *)colors
 {
-    [self setPopUpViewAnimatedColors:colors withPositions:nil];
+    if ([colors count]==0) {
+        return;
+    } else if ([colors count]==1) {
+        [self setPopUpViewAnimatedColors:@[colors[0], colors[0]] withPositions:@[@(0.0), @(1.0)]];
+    } else {
+        CGFloat spacer = 1.0 / ([colors count]-1);
+        NSMutableArray *positions = [[NSMutableArray alloc] initWithCapacity:[colors count]];
+        for (NSInteger i = 0; i < [colors count]; i++) {
+            [positions addObject:@(i * spacer)];
+        }
+        [positions addObject:@(1.0)];
```

Comment:
Hmm… that's weird, sorry about that. 

In the example project the gradient animation works without having to manually calculate the gradient stop positions. Are you doing something slightly different to the example project, which could potentially result in the black background instead of a gradient? Perhaps there's something unclear in the API that could be the cause?